packages/babel-core/src/config/helpers/config-api.ts

```diff
@@ -24,7 +24,7 @@ type CallerFactory = (
   extractor: (callerMetadata: CallerMetadata | void) => unknown,
 ) => SimpleType;
 type TargetsFunction = () => Targets;
-type AssumptionFunction = (name: string) => boolean | void;
+type AssumptionFunction = (name: AssumptionName) => boolean;
```

Comment:
Technically a plugin could be used with an old `@babel/core` version that doesn't know about a new assumption and thus returns undefined.

We could annotate this as `(name: AssumptionName) => boolean | void`: it's fine (and more helpful) if the parameter is more restrictive, but the return type should cover all the possibilities to avoid possible bugs caused by assuming that it's always a boolean.

Reply:
If we stick to the principle that an assumption always introduce certain non-spec constraints, I would not oppose to make it always return `false` for unknown assumptions. Otherwise plugins has to cast `.assumption(name)` to boolean by themselves.

Reply:
Well, there is at least `noNewArrow` that does `api.assumption("noNewArrow") ?? true`.

Reply:
Don't we want to default it to `false` in Babel 8? 🙂